pilot/test/integration/infra.go

```diff
@@ -110,6 +133,49 @@ const (
 	ConfigMapKey = "mesh"
 )
 
+func newInfra(name string) *infra {
+	return &infra{
+		Name:           name,
+		KubeConfig:     "",
+		TestDataDir:    "pilot/test/integration/testdata/",
+		CertDir:        "pilot/docker/certs/",
+		Hub:            "gcr.io/istio-testing",
+		Tag:            "",
+		Namespace:      "",
+		IstioNamespace: "",
+		Registry:       string(serviceregistry.KubernetesRegistry),
+		Verbosity:      2,
+		Auth:           meshconfig.MeshConfig_NONE,
+		MixerCustomConfigFile: mixerConfigFile,
+		PilotCustomConfigFile: pilotConfigFile,
+		Mixer:                 true,
+		Ingress:               true,
+		Zipkin:                true,
+		DebugPort:             0,
+		SkipCleanup:           false,
+		SkipCleanupOnFailure:  false,
+		CheckLogs:             false,
+		ErrorLogsDir:          "",
+		CoreFilesDir:          "",
+		TestCount:             1,
+		SelectedTest:          "",
+		debugImagesAndMode:    true,
+		UseAutomaticInjection: false,
+		UseAdmissionWebhook:   false,
+		AdmissionServiceName:  "istio-pilot",
+	}
+}
+
+func (infra *infra) copyWithDefaultAuth() *infra {
```

Comment:
The function name implies something is copied, but the code is mutating the `infra` object and returning it. Maybe something like the following?

```golang
out := *infra
out.Name = ....
return &out
```